pkg/offsetguess/offsetguess.go

```diff
@@ -0,0 +1,293 @@
+// +build linux
+
+package offsetguess
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/iovisor/gobpf/elf"
+
+	"github.com/weaveworks/tcptracer-bpf/pkg/byteorder"
+)
+
+type tcpTracerState uint64
+
+const (
+	uninitialized tcpTracerState = iota
+	checking
+	checked
+	ready
+)
+
+type guessWhat uint64
+
+const (
+	guessSaddr guessWhat = iota
+	guessDaddr
+	guessFamily
+	guessSport
+	guessDport
+	guessNetns
+	guessDaddrIPv6
+)
+
+type tcpTracerStatus struct {
```

Comment:
Oh, I see what you mean now.

Empiric tests[1] show that the padding and alignment are the same in the C structure, at least on my system.

However, one thing we could do to be extra-sure is define this structure in C and let cgo handle that. A not-cleaned-up-yet version of this can be found in [iaguis/tcptracer-bpf-Cstruct](https://github.com/kinvolk/tcptracer-bpf/commit/121333edbb0af79809ced17e77b22b6d4776c6a8).

Let me know what you think. Thanks!

[1] The fact that it worked fine for us shows that there're no alignment issues. Also, I defined the struct in C in on a test go program, run cgo on it to get the generated Go types and compared that with our pure Go type (specifically I checked the addresses of the fields with gdb to see where the fields started)